src/math/SeparatingAxisTriangle.js

```diff
@@ -159,126 +159,163 @@ SeparatingAxisTriangle.prototype.intersectsTriangle = ( function () {
+				const sb = satBounds2[ i ];
+				const sa = satAxes2[ i ];
+				cachedSatBounds.setFromPoints( sa, arr1 );
+				if ( sb.isSeparated( cachedSatBounds ) ) return false;
 
 			}
 
-		}
+			// check crossed axes
+			for ( let i = 0; i < 4; i ++ ) {
+
+				const sa1 = satAxes1[ i ];
+				for ( let i2 = 0; i2 < 4; i2 ++ ) {
 
-		if ( target ) {
+					const sa2 = satAxes2[ i2 ];
+					cachedAxis.crossVectors( sa1, sa2 );
+					cachedSatBounds.setFromPoints( cachedAxis, arr1 );
+					cachedSatBounds2.setFromPoints( cachedAxis, arr2 );
+					if ( cachedSatBounds.isSeparated( cachedSatBounds2 ) ) return false;
+
+				}
 
-			const plane1 = this.plane;
-			const plane2 = other.plane;
+			}
 
-			if ( Math.abs( plane1.normal.dot( plane2.normal ) ) > 1.0 - 1e-10 ) {
+			if ( target ) {
 
 				// TODO find two points that intersect on the edges and make that the result
 				console.warn( 'SeparatingAxisTriangle.intersectsTriangle: Triangles are coplanar which does not support an output edge. Setting edge to 0, 0, 0.' );
+
 				target.start.set( 0, 0, 0 );
 				target.end.set( 0, 0, 0 );
 
-			} else {
+			}
 
-				// find the edge that intersects the other triangle plane
-				const points1 = this.points;
-				let found1 = false;
-				for ( let i = 0; i < 3; i ++ ) {
+			return true;
 
-					const p1 = points1[ i ];
-					const p2 = points1[ ( i + 1 ) % 3 ];
+		} else {
 
-					edge.start.copy( p1 );
-					edge.end.copy( p2 );
+			// find the edge that intersects the other triangle plane
+			const points1 = this.points;
+			let found1 = false;
+			let count1 = 0;
+			for ( let i = 0; i < 3; i ++ ) {
 
-					if ( plane2.intersectLine( edge, found1 ? edge1.start : edge1.end ) ) {
+				const p1 = points1[ i ];
+				const p2 = points1[ ( i + 1 ) % 3 ];
 
-						if ( found1 ) {
+				edge.start.copy( p1 );
+				edge.end.copy( p2 );
 
-							break;
+				if ( plane2.intersectLine( edge, found1 ? edge1.start : edge1.end ) ) {
 
-						}
+					count1 ++;
+					if ( found1 ) {
 
-						found1 = true;
+						break;
 
 					}
 
+					found1 = true;
+
 				}
 
-				// find the other triangles edge that intersects this plane
-				const points2 = other.points;
-				let found2 = false;
-				for ( let i = 0; i < 3; i ++ ) {
+			}
+
+			if ( count1 !== 2 ) {
 
-					const p1 = points2[ i ];
-					const p2 = points2[ ( i + 1 ) % 3 ];
+				return false;
+
+			}
 
-					edge.start.copy( p1 );
-					edge.end.copy( p2 );
+			// find the other triangles edge that intersects this plane
+			const points2 = other.points;
+			let found2 = false;
+			let count2 = 0;
+			for ( let i = 0; i < 3; i ++ ) {
 
-					if ( plane1.intersectLine( edge, found2 ? edge2.start : edge2.end ) ) {
+				const p1 = points2[ i ];
+				const p2 = points2[ ( i + 1 ) % 3 ];
 
-						if ( found2 ) {
+				edge.start.copy( p1 );
+				edge.end.copy( p2 );
 
-							break;
+				if ( plane1.intersectLine( edge, found2 ? edge2.start : edge2.end ) ) {
```

Comment:
Careful, if line is coplanar with plane, intersectLine returns the line point, which could lead to a wrong intersection, right?

Reply:
Thanks. I added a case to check for coplanar lines:

![image](https://user-images.githubusercontent.com/734200/158437754-08910097-e141-4f5b-b441-d07c0cc9b8e9.png)